config-model/src/main/java/com/yahoo/vespa/model/application/validation/change/RestartOnDeployForOnnxModelChangesValidator.java

```diff
@@ -94,16 +100,38 @@ private Optional<String> modelChanged(OnnxModelCost.ModelInfo a, OnnxModelCost.M
     }
 
     private static void setRestartOnDeployAndAddRestartAction(List<ConfigChangeAction> actions, ApplicationContainerCluster cluster, String message) {
+        log.log(INFO, message);
         cluster.onnxModelCostCalculator().setRestartOnDeploy();
         actions.add(new VespaRestartAction(cluster.id(), message));
     }
 
-    private static boolean enoughMemoryToAvoidRestart(ApplicationContainerCluster cluster) {
-        // Node memory is known so convert available memory percentage to node memory percentage
+    private static boolean enoughMemoryToAvoidRestart(ApplicationContainerCluster clusterInCurrentModel,
+                                                      ApplicationContainerCluster cluster,
+                                                      DeployLogger deployLogger) {
+        double currentModelCostInGb = onnxModelCostInGb(clusterInCurrentModel);
+        double nextModelCostInGb = onnxModelCostInGb(cluster);
+
         double totalMemory = cluster.getContainers().get(0).getHostResource().realResources().memoryGb();
-        double availableMemory = Math.max(0, totalMemory - Host.memoryOverheadGb);
-        double costInGb = (double) cluster.onnxModelCostCalculator().aggregatedModelCostInBytes() / 1024 / 1024 / 1024;
-        return ( 2 * costInGb < availableMemory);
+        double availableMemory = Math.max(0, totalMemory - Host.memoryOverheadGb - currentModelCostInGb - currentModelCostInGb);
```

Comment:
Shouldn't one of these `currentModelCostInGb` be `nextModelCostInGb`?

Reply:
Yes, thanks, fixed in https://github.com/vespa-engine/vespa/pull/29682